airbyte-integrations/connectors/destination-mssql-v2/src/main/kotlin/io/airbyte/integrations/destination/mssql/v2/convert/ResultSetToAirbyteValue.kt

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.mssql.v2.convert
+
+import io.airbyte.cdk.load.data.AirbyteValue
+import io.airbyte.cdk.load.data.ArrayType
+import io.airbyte.cdk.load.data.ArrayTypeWithoutSchema
+import io.airbyte.cdk.load.data.ArrayValue
+import io.airbyte.cdk.load.data.BooleanType
+import io.airbyte.cdk.load.data.BooleanValue
+import io.airbyte.cdk.load.data.DateType
+import io.airbyte.cdk.load.data.DateValue
+import io.airbyte.cdk.load.data.IntegerType
+import io.airbyte.cdk.load.data.IntegerValue
+import io.airbyte.cdk.load.data.NullValue
+import io.airbyte.cdk.load.data.NumberType
+import io.airbyte.cdk.load.data.NumberValue
+import io.airbyte.cdk.load.data.ObjectType
+import io.airbyte.cdk.load.data.ObjectTypeWithEmptySchema
+import io.airbyte.cdk.load.data.ObjectTypeWithoutSchema
+import io.airbyte.cdk.load.data.ObjectValue
+import io.airbyte.cdk.load.data.StringType
+import io.airbyte.cdk.load.data.StringValue
+import io.airbyte.cdk.load.data.TimeTypeWithTimezone
+import io.airbyte.cdk.load.data.TimeTypeWithoutTimezone
+import io.airbyte.cdk.load.data.TimeValue
+import io.airbyte.cdk.load.data.TimestampTypeWithTimezone
+import io.airbyte.cdk.load.data.TimestampTypeWithoutTimezone
+import io.airbyte.cdk.load.data.TimestampValue
+import io.airbyte.cdk.load.data.UnionType
+import io.airbyte.cdk.load.data.UnknownType
+import io.airbyte.integrations.destination.mssql.v2.MSSQLQueryBuilder
+import io.airbyte.integrations.destination.mssql.v2.MSSQLQueryBuilder.NamedValue
+import io.airbyte.protocol.models.Jsons
+import java.sql.ResultSet
+import java.sql.Timestamp
+import java.time.LocalTime
+import java.time.OffsetDateTime
+import java.time.format.DateTimeFormatter
+
+class ResultSetToAirbyteValue {
+    companion object {
+        fun ResultSet.getAirbyteNamedValue(field: MSSQLQueryBuilder.NamedField): NamedValue =
+            when (field.type.type) {
+                is StringType -> getStringValue(field.name)
+                is ArrayType -> getArrayValue(field.name)
+                ArrayTypeWithoutSchema -> getArrayValue(field.name)
```

Comment:
More mixes of `is type` and just `type`.  If both work, we should be consistent here and choose one approach.

Reply:
This is because we are mixing two concepts, `AirbyteValue` is a sealed interface and we are mixing `data object` and `value class`. `is` refers to `value class`, cases where `is` is omitted are `data object`.
That's also how intellij auto generates the support all cases for a sealed interface, at least with the default style config.